wgengine/userspace.go

```diff
@@ -102,8 +103,9 @@ func NewUserspaceEngine(logf logger.Logf, tunname string, listenPort uint16) (En
 		return nil, err
 	}
 	logf("CreateTUN ok.")
+	tun := wgtun.WrapTun(logf, tundev)
 
-	e, err := NewUserspaceEngineAdvanced(logf, tundev, router.New, listenPort)
+	e, err := NewUserspaceEngineAdvanced(logf, tun, router.New, listenPort)
```

Comment:
we should use a consistent variable name for the interface value vs concrete value. Below it's tundev for the concrete type.